### ISSUE TITLE
Getting "Error exception of type 'google.googlesignin+signin exception' was thrown" error

### ISSUE THREAD

**Issue by surajsk1812** [googlesamples/google-signin-unity#184]
Normal Google login is working but silent google login is not working.
Getting "Error exception of type 'google.googlesignin+signin exception' was thrown" error

**Comment by DellaBitta** [comment]
Google Sign In Team, please see this [Issue #1050 in the Firebase Quickstart](https://github.com/firebase/quickstart-unity/issues/1050) for more user information about this problem.

**Comment by winocur** [comment]
Having the same issue on android. Only in some devices. Error doesn't give any details and logcat has nothing else suspicious. Need help ASAP.

**Comment by surajsk1812** [comment]
@winocur Let me know if you resolve this bug. Google SignIn Team not responding here

**Comment by yjh0502** [comment]
Problem might be: https://stackoverflow.com/questions/47632035/google-sign-in-error-12500

**Comment by hdev72** [comment]
resolved
Having the same issue because of different package name with firebase project.

**Comment by tdx110** [comment]
Problem solved.
I wrote everywhere I can about the solution.
APK and AAB file have different SHA1 and SHA256.
In Firebase, it's best to add SHA for both applications.
You can check the SHA from the AAB file in Google Play Console -> Your application -> Configuration -> Application integrity -> Application signing.
I added these SHA keys to Firebase and the problem was gone.

**Comment by NGDtuanh** [comment]
> Problem solved. I wrote everywhere I can about the solution. APK and AAB file have different SHA1 and SHA256. In Firebase, it's best to add SHA for both applications. You can check the SHA from the AAB file in Google Play Console -> Your application -> Configuration -> Application integrity -> Application signing. I added these SHA keys to Firebase and the problem was gone.

your are my savior. thank you very much

**Comment by jitendra-unity-vsf** [comment]
**Problem sovled**

**Get SHA-1 and SHA-256 from the .aab file uploaded on the play store** 
Google Play Console -> Your application -> Configuration -> Application integrity -> Application signing

**Get SHA-1 and SHA-256 using following command from the keystore you are using while preparing build** 
keytool -list -v -keystore "path\debug.keystore" -alias aliasname -storepass keystorepassword -keypass aliaspassword

**Run above command where your JDK installed and keytool.exe placed.** 
i.e. C:\Program Files\Unity\Hub\Editor\2020.3.30f1\Editor\Data\PlaybackEngines\AndroidPlayer\OpenJDK\bin

**Once you have SHA keys from .aab and keystore, then add those under the firebase app setting**

![Screenshot 2023-12-21 145306](https://github.com/googlesamples/google-signin-unity/assets/110443464/128dfb1e-1101-4a97-b1fc-74259478ba3d)

**Comment by browseroffice70-cmd** [comment]
We got this for Unity app with Firebase - Google authentication

**Error:**

- Google Sign-In Error: One or more errors occurred. (There was an error while trying to get your package certificate hash.) (There was an error while trying to get your package certificate hash.)
UnityEngine.DebugLogHandler:Internal_Log(LogType, LogOption, String, Object)

- System.AggregateException: One or more errors occurred. (Exception of type 'Google.GoogleSignIn+SignInException' was thrown.) ---> Google.GoogleSignIn+SignInException: Exception of type 'Google.GoogleSignIn+SignInException' was thrown.

- Debug TokenPendingResult  Calling onResult for callback. result: Status: Status{statusCode=DEVELOPER_ERROR,   resolution=null} <null>

ALL OF THESE POINTED THE SAME(For different google sign-in approaches we tried)


**FIX:**
We used Google SignIn Legacy package(https://github.com/googlesamples/google-signin-unity)
We had to add the Legacy App signing key certificate SHA keys to firebase app fingerprint along with latest App signing key certificate.This worked for me.

<img width="1638" height="690" alt="Image" src="https://github.com/user-attachments/assets/4119d197-2593-45d5-a335-388349353041" />
